src/k2/cpo/client/Client.h

```diff
@@ -69,14 +69,27 @@ class CPOClient {
             if (status == Statuses::S403_Forbidden || status.is2xxOK()) {
                 Duration s = std::min(deadline.getRemaining(), cpo_request_backoff());
                 return seastar::sleep(s).then([this, name, deadline]() -> seastar::future<Status> {
-                    return _getAssignedPartitionWithRetry(deadline, name, dto::Key{});
+                    return _checkAllParititonsAssigned(deadline, name)
+                    .then([] (auto &&status) {
+                        if (status.is2xxOK()) {
+                            return seastar::make_ready_future<Status>(Statuses::S201_Created("Collection created"));
+                        }
+                        return seastar::make_ready_future<Status>(status);
+                    });
                 });
             }
 
             return seastar::make_ready_future<Status>(std::move(status));
         });
     }
 
+
```

Comment:
no need for this public method. i think we can just make _getPartitionGetterForCollection public and rename it